ij/src/main/kotlin/org/jetbrains/academy/test/system/test/BaseIjTestClass.kt

```diff
@@ -1,8 +1,55 @@
 package org.jetbrains.academy.test.system.test
 
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiRecursiveElementWalkingVisitor
 import com.intellij.testFramework.fixtures.BasePlatformTestCase
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtProperty
 
 /**
  * A Base test class to create tests with PSI without adding intellij dependency into courses directly.
  */
-open class BaseIjTestClass : BasePlatformTestCase()
+open class BaseIjTestClass : BasePlatformTestCase() {
+
+    fun findMethodsWithContent(content: String): List<String> {
```

Comment:
It does not work, students can use different formatting etc. You need to create a psi by content and use CodeFormatter to reformat code, Also users can create nested functions and in this case `element.text.contains(content)` does not work, you need to separate the function body - you can find the body (like I showed in getConstValue function) and compare it directly

Reply:
Also I'd suggest to move these functions into src and here just call them with fixture. Also, please ad descriptions

Reply:
Below you suggest putting it in a new class that implements BaseIjTestClass - https://github.com/jetbrains-academy/kotlin-test-framework/pull/18#discussion_r1269489699. It seems to conflict with this comment. What did you mean?